docs/docsite/rst/dev_guide/developing_python_3.rst

```diff
@@ -4,47 +4,29 @@
-3. shared ``module_utils`` code - the common code that's used by modules to perform tasks and sometimes used by controller-side code as well
+1. controller-side code - code that runs on the machine where you invoke :command:`/usr/bin/ansible`, only needs to support the controller's Python versions
+2. modules - the code which Ansible transmits to and invokes on the managed machine. They need to support the 'managed node' Python versions, with some exceptions.
+3. shared ``module_utils`` code - the common code that's used by modules to perform tasks and sometimes used by controller-side code as well, so they need to support the same range of Python as the modules.
 
 However, the three types of code do not use the same string strategy. If you're developing a module or some ``module_utils`` code, be sure to read the section on string strategy carefully.
 
+.. note:
+    - While modules can be written in any language, the above applies to code contributed to the core project, which only supports specific Python versions (and Powershell for Windows)
+
 Minimum version of Python 3.x and Python 2.x
 ============================================
 
-On the controller we support Python 3.5 or greater and Python 2.7 or greater.  Module-side, we
-support Python 3.5 or greater and Python 2.6 or greater.
-
-Python 3.5 was chosen as a minimum because it is the earliest Python 3 version adopted as the
-default Python by a Long Term Support (LTS) Linux distribution (in this case, Ubuntu-16.04).
-Previous LTS Linux distributions shipped with a Python 2 version which users can rely upon instead
-of the Python 3 version.
-
-For Python 2, the default is for modules to run on at least Python 2.6.  This allows
-users with older distributions that are stuck on Python 2.6 to manage their
-machines.  Modules are allowed to drop support for Python 2.6 when one of
-their dependent libraries requires a higher version of Python. This is not an
-invitation to add unnecessary dependent libraries in order to force your
-module to be usable only with a newer version of Python; instead it is an
-acknowledgment that some libraries (for instance, boto3 and docker-py) will
-only function with a newer version of Python.
-
-.. note:: Python 2.4 Module-side Support:
+See  ../../installation_guide/intro_installation.html#control-node-requirements and  ../../installation_guide/intro_installation.html#control-node-requirements for the
+specific versions supported.
 
-    Support for Python 2.4 and Python 2.5 was dropped in Ansible-2.4.  RHEL-5
-    (and its rebuilds like CentOS-5) were supported until April of 2017.
-    Ansible-2.3 was released in April of 2017 and was the last Ansible release
-    to support Python 2.4 on the module-side.
+Your custom modules can support any version of Python (or other languages) you want, but the above are the requirements for the code contributed to the Ansible project.
```

Comment:
to the Ansible project to to ansible-core?

Reply:
afaik, both?